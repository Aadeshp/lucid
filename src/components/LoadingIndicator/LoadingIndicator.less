.lucid-LoadingIndicator {

	position: relative;

	.lucid-Icon {
		display: inline-block;
	}

	&-message-container {
		.transition-group-animation-fade();
		position: absolute;
		top: 0;
		right: 0;
		bottom: 0;
		left: 0;
		display: flex;
		flex-direction: row;
		align-items: center;
		justify-content: center;
		background: none;
	}

	&-has-overlay-dark {
		background-color: @color-gray-60;
	}

	&-has-overlay-light {
		background-color: @color-white-85;
	}
}
