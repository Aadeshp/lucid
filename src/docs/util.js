import _ from 'lodash';
import { markdown } from 'markdown';

export function toMarkdown(md) {
	if (md) {
		return {
			__html: markdown.toHTML(md),
		};
	}

	return {
		__html: '',
	};
}

export function handleHighlightCode() {
	if (window.hljs) {
		//eslint-disable-line
		_.each(document.querySelectorAll('pre code'), block => {
			hljs.highlightBlock(block); //eslint-disable-line
		});
	}
}

export function sanitizeExamplePath(path) {
	return _.flow(
		x => _.split(x, '/'),
		x => _.compact(x),
		x => _.drop(x),
		x => _.join(x, '-'),
		x => _.split(x, '.'),
		x => _.join(x, '-')
	)(path);
}

export function stripIndent(text) {
	if (!text) {
		return '';
	}

	const sigLines = text.split(/\n/).filter(line => !/^\s*$/.test(line));
	const leadingWhitespaces = sigLines.map(line => /^(\s*)/g.exec(line)[1]);
	const minimumLeadingWhitespace = leadingWhitespaces.reduce(
		(min, line) => (line.length < min.length ? line : min),
		leadingWhitespaces[0]
	);
	return text
		.split(/\n/)
		.map(line => line.replace(minimumLeadingWhitespace, ''))
		.join('\n');
}
