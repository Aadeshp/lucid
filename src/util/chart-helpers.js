import _ from 'lodash';
import d3TimeFormat from 'd3-time-format';

/**
 * groupByFields
 *
 * This will return the data in a similar format to d3Shape.stack
 * but without the stacking of the data.
 *
 * @param {object[]} collection - normalized data you want to operate on
 * @param {string[]} fields - fields to pluck off for the y data
 * @return {array[]} - array of arrays, one for each field
 */
export function groupByFields(collection, fields) {
	const fieldsArray = [].concat(fields);

	return _.map(fieldsArray, (field) => {
		return _.map(collection, (item) => item[field]);
	});
}

/**
 * toPairs
 *
 * Takes a collection of data and pairs up some x and y data.
 *
 * @param {object[]} collection
 * @param {string} xField
 * @param {string} yField
 * @return {array} - E.g. [[x, y], [x, y]]
 */
export function toPairs(collection, xField, yField) {
	return _.map(collection, (item) => [item[xField], item[yField]]);
}

/**
 * byFields
 *
 * Takes a collection of data and returns an array of all the fields off that
 * collection.
 *
 * @param {object[]} collection
 * @param {string[]} fields
 * @return {array}
 */
export function byFields(collection, fields) {
	const fieldArray = [].concat(fields);

	return _.reduce(fieldArray, (acc, field) => {
		return acc.concat(_.map(collection, field));
	}, []);
}

/**
 * minByFields
 *
 * Returns the minimum element from a collection by a set of fields.
 *
 * @param {object[]} collection
 * @param {string[]} fields
 * @return {any}
 */
export function minByFields(collection, fields) {
	return _.min(byFields(collection, fields));
}

/**
 * maxByFields
 *
 * Returns the maximum element from a collection by a set of fields.
 *
 * @param {object[]} collection
 * @param {string[]} fields
 * @return {any}
 */
export function maxByFields(collection, fields) {
	return _.max(byFields(collection, fields));
}

/**
 * maxByFieldsStacked
 *
 * Returns the max sum of a set of fields from a collection
 *
 * @param {object[]} collection
 * @param {string[]} fields
 * @return {any}
 */
export function maxByFieldsStacked(collection, fields) {
	const fieldArray = [].concat(fields);

	const sums = _.reduce(collection, (acc, item) => {
		const sum = _.chain(item)
			.pick(fieldArray)
			.toArray()
			.sum()
			.value();
		return acc.concat(sum);
	}, []);

	return _.max(sums);
}

/**
 * discreteTicks
 *
 * Returns `count` evenly spaced, representative values from the `array`.
 *
 * @param {array} array
 * @param {number} size - should be greater than 1
 * @return {array}
 */
export function discreteTicks(array, count) {
	if (!array || _.isNil(count) || array.length <= count) {
		return array;
	}

	const step = (array.length - 1) / Math.max(1, count - 1);

	return _.reduce(_.times(count), (acc, n) => {
		return acc.concat(array[Math.round(n  * step)]);
	}, []);
}

/**
 * transformFromCenter
 *
 * Scaling paths from their center is tricky. This function
 * helps do that be generating a translate/scale transform
 * string with the correct numbers.
 *
 * @param {number} x - the x data point where you want the path to be centered at
 * @param {number} y - the y data point where you want the path to be centered at
 * @param {number} xCenter - the x coordinate of the center of the path you're trying to transform
 * @param {number} yCenter - the x coordinate of the center of the path you're trying to transform
 * @param {number} scale - number to scale to, 2 would be 2x bigger
 * @return {string} - transform string
 */
export function transformFromCenter(x, y, xCenter, yCenter, scale) {
	return `translate(${((1 - scale) * xCenter) + (x - xCenter)}, ${((1 - scale) * yCenter) + (y - yCenter)}) scale(${scale})`;
}

const formatMillisecond = d3TimeFormat.timeFormat('.%L');
const formatSecond = d3TimeFormat.timeFormat(':%S');
const formatMinute = d3TimeFormat.timeFormat('%I:%M');
const formatHour = d3TimeFormat.timeFormat('%I %p');
const formatDay = d3TimeFormat.timeFormat('%a %d');
const formatWeek = d3TimeFormat.timeFormat('%b %d');
const formatMonth = d3TimeFormat.timeFormat('%B');
const formatYear = d3TimeFormat.timeFormat('%Y');

export function multiFormat(date) {
	return (d3TimeFormat.timeSecond(date) < date ? formatMillisecond
		: d3TimeFormat.timeMinute(date) < date ? formatSecond
		: d3TimeFormat.timeHour(date) < date ? formatMinute
		: d3TimeFormat.timeDay(date) < date ? formatHour
		: d3TimeFormat.timeMonth(date) < date ? (d3TimeFormat.timeWeek(date) < date ? formatDay : formatWeek)
		: d3TimeFormat.timeYear(date) < date ? formatMonth
		: formatYear)(date);
}

