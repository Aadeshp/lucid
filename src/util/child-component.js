import React from 'react';
import _ from 'lodash';

/**
 * createChildComponent
 *
 * Returns a React Component which does not render anything, but has some useful static functions
 *
 * @return {ReactComponent} which renders `null` but exposes static helper functions
 */
export function createChildComponent (definition={}) {
	return React.createClass({
		propTypes: _.get(definition, 'propTypes', null),
		displayName: _.get(definition, 'displayName', null),
		render: () => null,
		statics: {
			isRenderNull: true,

			/**
			 * findInChildren
			 *
			 * Returns all elements with this component type
			 *
			 * @param {array} elements - usually `this.props.children` but can be any list of elements which might have this type
			 * @return {array} elements with this component type
			 */
			findInChildren(elements) {
				let component = this;
				let ownElements = [];
				React.Children.forEach(elements, (element) => {
					if (element && element.type === component) {
						ownElements.push(element);
					}
				});
				return ownElements;
			},

			/**
			 * findInChildrenAsProps
			 *
			 * Returns the `props` object of each element with this component type
			 *
			 * @param {array} elements - usually `this.props.children` but can be any list of elements which might have this type
			 * @return {array} `props` objects of elements with this component type
			 */
			findInChildrenAsProps(elements) {
				let component = this;
				return _.map(component.findInChildren(elements), 'props');
			},

			/**
			 * findInProps
			 *
			 * Given a set of parent props this will return only the props for this
			 * component type
			 *
			 * @param {object} parentProps - usually `this.props` that may contain a prop for the current component type
			 * @return {array} - an array of normalized props
			 */
			findInProps(parentProps) {
				// Extract out the defined `childProps` from the parent class
				const currentProps = _.chain(parentProps)
					.get(definition.propName) // grab the prop we care about, e.g. "Child"
					.thru(x => [x]) // wrap in an array
					.flatten()
					.filter(x => x) // remove falsey values
					.value();

				return _.map(currentProps, (childProp) => {
					if (_.isPlainObject(childProp)) {
						return childProp;
					} else {
						return {
							children: childProp
						};
					}
				});
			},

			/**
			 * findInAllAsProps
			 *
			 * Given a parent props object, this will return a mashup of all the
			 * children and/or props for the current component type. This is
			 * particularly useful for providing a flexible component API. It allows
			 * consumers to either pass stuff through props or children.
			 *
			 * @param {object} parentProps - usually `this.props` that may contain a prop for the current component type
			 * @return {array} - an array of `props`
			 */
			findInAllAsProps(parentProps) {
				let component = this;

				const propsFromProps = component.findInProps(parentProps);
				const propsFromChildren = component.findInChildrenAsProps(parentProps.children);

				return propsFromProps.concat(propsFromChildren);
			}
		}
	});
}

export function rejectNullElements(children) {
	return _.reject(React.Children.toArray(children), (node) => (React.isValidElement(node) && node.type.isRenderNull));
}
