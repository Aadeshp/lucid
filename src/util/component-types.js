import React from 'react';
import _ from 'lodash';

// creates a React component
export function createClass(definition={}) {
	const {
		_isPrivate = false,
		statics = {},
		components = {},
		reducers = {},
		selectors = {},
		propName = null,
		propTypes = {},
		render = () => null,
		...restDefinition
	} = definition;

	const newDefinition = {
		...restDefinition,
		statics: {
			...statics,
			...components,
			_isPrivate,
			reducers,
			selectors,
			propName,
		},
		propTypes: _.assign({}, propTypes, _.mapValues(definition.components, () => React.PropTypes.any)),
		render,
	};

	newDefinition.statics.definition = newDefinition;

	return React.createClass(newDefinition);
}

// return all elements matching the specified types
export function filterTypes(children, types=[]) {
	types = [].concat(types); // coerce to Array

	return _.filter(
		React.Children.toArray(children),
		(element) => (React.isValidElement(element) && _.includes(types, element.type))
	);
}

// return all elements not matching the specified types
export function rejectTypes(children, types=[]) {
	types = [].concat(types); // coerce to Array

	return _.reject(
		React.Children.toArray(children),
		(element) => (React.isValidElement(element) && _.includes(types, element.type))
	);
}

// return an array of elements (of the given type) for each of the values
export function createElements(type, values=[]) {
	return _.reduce([].concat(values), (elements, typeValue) => {
		if (React.isValidElement(typeValue) && typeValue.type === type) {
			return elements.concat(typeValue);
		} else if (_.isPlainObject(typeValue) && !React.isValidElement(typeValue)) {
			return elements.concat(React.createElement(type, typeValue));
		} else if (_.isUndefined(typeValue)) {
			return elements;
		} else {
			return elements.concat(React.createElement(type, null, typeValue));
		}
	}, []);
}

// return all elements found in props and children of the specified types
export function findTypes(props, types=[]) {
	// get elements from props (using type.propName)
	const elementsFromProps = _.reduce(_.castArray(types), (acc, type) => {
		if (!_.isUndefined(type.propName)) {
			const propMatches = _.flatten(_.values(_.pick(props, type.propName)));
			return acc.concat(createElements(type, propMatches));
		}
		return acc;
	}, []);

	// return elements from props and elements from children
	return elementsFromProps.concat(filterTypes(props.children, types));
}

// return the first element found in props and children of the specificed type(s)
export function getFirst(props, types, defaultValue) {
	return _.first(findTypes(props, types)) || defaultValue;
}

// omit props defined in propTypes of the given type
// and any extra keys given in third argument
export function omitProps(props, type, keys = []) {
	return _.omit(props, _.keys(type.propTypes).concat(keys).concat(['initialState']));
}
