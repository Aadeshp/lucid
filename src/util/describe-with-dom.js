function describeWithDOM(name, func) {
	describe(`${name} (functional tests using ${window.realOrFake})`, func);
}

describeWithDOM.only = function (name, func) {
	describe.only(`${name} (functional tests using ${window.realOrFake})`, func);
};

export default describeWithDOM;
