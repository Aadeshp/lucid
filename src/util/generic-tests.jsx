import sinon from 'sinon';
import React from 'react';
import { mount, shallow } from 'enzyme';
import assert from 'assert';
import describeWithDOM  from './describe-with-dom';
import _ from 'lodash';
import * as lucid from '../index';

// Common tests for all our components
export function common(Component, getDefaultProps=_.noop) {

	function generateDefaultProps(props={}) {
		return _.assign({}, getDefaultProps(), props);
	}

	describe(`[common]`, () => {
		it('should have a `displayName` defined', () => {
			assert(Component.displayName);
		});

		it('should pass through styles to the root element', () => {
			let style = {
				backgroundColor: '#f0f'
			};
			const wrapper = shallow(<Component {...generateDefaultProps()} style={style}/>);
			assert.deepEqual(wrapper.first().prop('style'), style);
		});

		it('should pass through `className`', () => {
			let expectedClass = 'rAnDoM';
			const wrapper = shallow(<Component {...generateDefaultProps()} className={expectedClass}/>);
			let classNames = wrapper.first().prop('className').split(' ');

			assert(_.includes(classNames, expectedClass), `'${classNames}' should include '${expectedClass}'`);
		});

		it('should have an application scoped base class', () => {
			let expectedClass = 'lucid-' + Component.displayName;
			const wrapper = shallow(<Component {...generateDefaultProps()} />);
			let classNames = wrapper.first().prop('className').split(' ');

			assert(_.includes(classNames, expectedClass), `'${classNames}' should include '${Component.displayName}'`);
		});

		it('should have only application scoped classes', () => {
			const wrapper = shallow(<Component {...generateDefaultProps()} />);
			let parentClasses = wrapper.first().prop('className').split(' ');
			let childrenClasses = wrapper.children().reduce((acc, node) => {
				if (!node.prop('className')) {
					return acc;
				}

				return acc.concat(node.prop('className').split(' '))
			}, []);

			let allClasses = parentClasses.concat(childrenClasses);

			assert(_.every(allClasses, (className) => {
				return _.includes(className, 'lucid-' + Component.displayName);
			}));
		});

		it('should only use onX convention for function proptypes', () => {
			assert(_.every(Component.propTypes, (value, key) => {
				// See the following tests from the React source code to figure out how
				// this works: https://github.com/facebook/react/blob/v0.14.7/src/isomorphic/classic/types/__tests__/ReactPropTypes-test.js
				let props = {};
				props[key] = _.noop;

				// Here we test the validation against a noop function, if it doesn't
				// have an error, then it's probably a function. The other possible is
				// that it's an `any`, so we'll test for that below
				let isProbablyFunction = !(value(props, key) instanceof Error);

				// We'll also test out a string to see if that valid
				props[key] = '';

				let isAny = isProbablyFunction && !(value(props, key) instanceof Error);

				// If it's probably a function, and it's not `any`, then we make sure
				// it starts with `on`
				if (isProbablyFunction && !isAny && !_.startsWith(key, 'on')) {
					return false
				}

				return true;
			}));
		});

		it('should be available as an exported module from index.js', () => {
			assert(lucid[Component.displayName]);
		});
	});
}

// Common tests for all our icon components
export function icons(Component) {
	describeWithDOM(`[icon]`, () => {
		it('should pass through isBadge prop to underlying Icon component', () => {
			const wrapper = mount(<Component isBadge={true} />);
			const classNames = wrapper.find('svg').prop('className').split(' ');
			const targetClassName = 'lucid-Icon-is-badge';
			assert(_.includes(classNames, targetClassName), `'${classNames}' should include '${targetClassName}'`);
		});
	});
}

// Common tests for all control components
export function controls(Component, { callbackName, controlSelector , eventType }) {
	// Use DOM tests here since some of our controls use dom events under the hood
	describeWithDOM('[control]', () => {
		it('should callback with `uniqueId` and `event` prop', () => {
			const expectedUniqueId = 32;
			const props = {
				uniqueId: expectedUniqueId,
				[callbackName]: sinon.spy(),
			};
			const wrapper = mount(<Component {...props} />);

			wrapper.find(controlSelector).simulate(eventType);

			// Last argument should be an object with `uniqueId` and `event`
			const { uniqueId, event } = _.last(props[callbackName].args[0]);

			assert(event, 'missing event');
			assert.equal(uniqueId, expectedUniqueId, 'incorrect or missing uniqueId');
		});
	});
}
