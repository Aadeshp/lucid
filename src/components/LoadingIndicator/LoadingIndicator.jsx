import React from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import { lucidClassNames } from '../../util/style-helpers';
import { createClass, getFirst, rejectTypes, omitProps } from '../../util/component-types';
import LoadingMessage from '../LoadingMessage/LoadingMessage';

const cx = lucidClassNames.bind('&-LoadingIndicator');

const {
	bool,
	node,
	oneOf,
	string,
} = React.PropTypes;

/**
 *
 * {"categories": ["communication"], "madeFrom": ["LoadingMessage"]}
 *
 * A loading indicator wrapper with optional overlay.
 *
 */
const LoadingIndicator = createClass({
	displayName: 'LoadingIndicator',
	propTypes: {
		/**
		 * Set this to `false` if you don't want the semi-transparent overlay over
		 * the wrapped content
		 */
		hasOverlay: bool,
		/**
		 * Class names that are appended to the defaults.
		 */
		className: string,
		/**
		 * Any valid React children.
		 */
		children: node,
		/**
		 * Controls the visibility of the `LoadingMessage` and overlay.
		 */
		isLoading: bool,
		/**
		 * Style variations for the overlay behind the loading indicator.
		 */
		overlayKind: oneOf([
			'light',
			'dark',
		]),
	},

	components: { LoadingMessage },

	getDefaultProps() {
		return {
			hasOverlay: true,
			isLoading: false,
			overlayKind: 'dark',
		};
	},

	render() {
		const {
			props,
			props: {
				hasOverlay,
				className,
				children,
				isLoading,
				overlayKind,
				...passThroughs,
			},
		} = this;

		const { LoadingMessage } = LoadingIndicator;

		const messageElement = getFirst(props, LoadingMessage, <LoadingMessage />);
		const otherChildren = rejectTypes(children, [LoadingMessage]);

		return (
			<div
				{...omitProps(passThroughs, LoadingIndicator)}
				className={cx('&', className)}
			>
				{otherChildren}
				<ReactCSSTransitionGroup
					transitionName={cx('&-message-container')}
					transitionEnterTimeout={300}
					transitionLeaveTimeout={300}
				>
					{isLoading && (
						<div className={cx('&-message-container', { [`&-has-overlay-${overlayKind}`]: hasOverlay })}>
							{messageElement}
						</div>
					)}
				</ReactCSSTransitionGroup>
			</div>
		);
	},
});

export default LoadingIndicator;
