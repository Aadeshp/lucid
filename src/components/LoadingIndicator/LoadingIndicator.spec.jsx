import React from 'react';
import { shallow } from 'enzyme';
import assert from 'assert';
import { common } from '../../util/generic-tests';
import LoadingIndicator from './LoadingIndicator';

const { LoadingMessage } = LoadingIndicator;

describe('LoadingIndicator', () => {
	common(LoadingIndicator);

	describe('isLoading', () => {

		describe('default', () => {

			let wrapper;

			beforeEach(() => {
				wrapper = shallow(
					<LoadingIndicator>
						<div>Some content</div>
					</LoadingIndicator>
				);
			});

			it('should not show loading message', () => {
				assert.equal(wrapper.find('.lucid-LoadingIndicator-message-container').length, 0);
				assert.equal(wrapper.find(LoadingMessage).length, 0);
			});

			it('should render other content', () => {
				assert(wrapper.contains(<div>Some content</div>));
			});

		});

		describe('false', () => {

			let wrapper;

			beforeEach(() => {
				wrapper = shallow(
					<LoadingIndicator isLoading={false}>
						<div>Some content</div>
					</LoadingIndicator>
				);
			});

			it('should not show loading message', () => {
				assert.equal(wrapper.find('.lucid-LoadingIndicator-message-container').length, 0);
				assert.equal(wrapper.find(LoadingMessage).length, 0);
			});

			it('should render other content', () => {
				assert(wrapper.contains(<div>Some content</div>));
			});

		});

		describe('true', () => {

			let wrapper;

			beforeEach(() => {
				wrapper = shallow(
					<LoadingIndicator isLoading>
						<div>Some content</div>
					</LoadingIndicator>
				);
			});

			it('should show loading message', () => {
				assert.equal(wrapper.find('.lucid-LoadingIndicator-message-container').length, 1);
				assert.equal(wrapper.find(LoadingMessage).length, 1);
			});

			it('should render other content', () => {
				assert(wrapper.contains(<div>Some content</div>));
			});

		});

	});

	describe('hasOverlay', () => {

		describe('default', () => {
			it('should add `&-has-overlay` class', () => {
				const wrapper = shallow(
					<LoadingIndicator isLoading>
						<div>Some content</div>
					</LoadingIndicator>
				);
				const messageContainer = wrapper.find('.lucid-LoadingIndicator-message-container');
				assert(messageContainer.hasClass('lucid-LoadingIndicator-has-overlay'));
			});
		});

		describe('true', () => {
			it('should add `&-has-overlay` class', () => {
				const wrapper = shallow(
					<LoadingIndicator
						isLoading
						hasOverlay
					>
						<div>Some content</div>
					</LoadingIndicator>
				);
				const messageContainer = wrapper.find('.lucid-LoadingIndicator-message-container');
				assert(messageContainer.hasClass('lucid-LoadingIndicator-has-overlay'));
			});
		});

		describe('false', () => {
			it('should not add `&-has-overlay` class', () => {
				const wrapper = shallow(
					<LoadingIndicator
						isLoading
						hasOverlay={false}
					>
						<div>Some content</div>
					</LoadingIndicator>
				);
				const messageContainer = wrapper.find('.lucid-LoadingIndicator-message-container');
				assert(!messageContainer.hasClass('lucid-LoadingIndicator-has-overlay'));
			});
		});

	});

	describe('custom loading message', () => {
		it('should use custom loading message', () => {
			const message = <LoadingMessage Title='Foo' Body='Bar'/>;
			const wrapper = shallow(
				<LoadingIndicator isLoading>
					<div>Some content</div>
					{message}
				</LoadingIndicator>
			);
			assert(wrapper.contains(message), 'must contain custom loading message');
		});
	});

	describe('overlayKind', () => {
		describe('light', () => {
			it('should add `&-kind-light` class', () => {
				const wrapper = shallow(
					<LoadingIndicator
						isLoading
						overlayKind='light'
					>
						<div>Some content</div>
					</LoadingIndicator>
				);
				const messageContainer = wrapper.find('.lucid-LoadingIndicator-message-container');
				assert(messageContainer.hasClass('lucid-LoadingIndicator-kind-light'));
			});
		});
	});

});
