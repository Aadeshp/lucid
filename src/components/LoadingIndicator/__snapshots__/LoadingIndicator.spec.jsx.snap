exports[`LoadingIndicator [common] example testing should match snapshot for 1.interactive 1`] = `
<LoadingIndicator
  hasOverlay={true}
  isLoading={true}
  overlayKind="dark">
  <Button
    hasOnlyIcon={false}
    isActive={false}
    isDisabled={false}
    onClick={[Function]}
    style={
      Object {
        "margin": 10,
      }
    }
    type="button">
    Get more data
  </Button>
  <Button
    hasOnlyIcon={false}
    isActive={false}
    isDisabled={false}
    onClick={[Function]}
    type="button">
    Switch overlay color
  </Button>
  <BarChart
    data={
      Array [
        Object {
          "x": "2015-01-01",
          "y": 0,
        },
        Object {
          "x": "2015-01-02",
          "y": 0,
        },
        Object {
          "x": "2015-01-03",
          "y": 0,
        },
        Object {
          "x": "2015-01-04",
          "y": 0,
        },
      ]
    }
    hasLegend={false}
    hasToolTips={true}
    height={400}
    margin={
      Object {
        "bottom": 50,
        "left": 80,
        "right": 20,
        "top": 10,
      }
    }
    palette={
      Array [
        "color-chart-0",
        "color-chart-1",
        "color-chart-2",
        "color-chart-3",
        "color-chart-4",
        "color-chart-5",
      ]
    }
    width={1000}
    xAxisField="x"
    xAxisFormatter={[Function]}
    xAxisTickCount={null}
    xAxisTitle={null}
    xAxisTitleColor="#000"
    yAxisFields={
      Array [
        "y",
      ]
    }
    yAxisIsStacked={false}
    yAxisMin={0}
    yAxisTickCount={null}
    yAxisTitle="Revenue"
    yAxisTitleColor="#000"
    yAxisTooltipFormatter={[Function]} />
</LoadingIndicator>
`;

exports[`LoadingIndicator [common] example testing should match snapshot for 2.default 1`] = `
<LoadingIndicator
  hasOverlay={true}
  isLoading={true}
  overlayKind="light">
  <BarChart
    data={
      Array [
        Object {
          "x": "2015-01-01",
          "y": 1,
        },
        Object {
          "x": "2015-01-02",
          "y": 2,
        },
        Object {
          "x": "2015-01-03",
          "y": 3,
        },
        Object {
          "x": "2015-01-04",
          "y": 5,
        },
      ]
    }
    hasLegend={false}
    hasToolTips={true}
    height={400}
    margin={
      Object {
        "bottom": 50,
        "left": 80,
        "right": 20,
        "top": 10,
      }
    }
    palette={
      Array [
        "color-chart-0",
        "color-chart-1",
        "color-chart-2",
        "color-chart-3",
        "color-chart-4",
        "color-chart-5",
      ]
    }
    width={1000}
    xAxisField="x"
    xAxisFormatter={[Function]}
    xAxisTickCount={null}
    xAxisTitle={null}
    xAxisTitleColor="#000"
    yAxisFields={
      Array [
        "y",
      ]
    }
    yAxisIsStacked={false}
    yAxisMin={0}
    yAxisTickCount={null}
    yAxisTitle={null}
    yAxisTitleColor="#000"
    yAxisTooltipFormatter={[Function]} />
</LoadingIndicator>
`;

exports[`LoadingIndicator [common] example testing should match snapshot for 3.custom-message 1`] = `
<div>
  <LoadingIndicator
    hasOverlay={true}
    isLoading={true}
    overlayKind="light">
    <BarChart
      data={
        Array [
          Object {
            "x": "2015-01-01",
            "y": 1,
          },
          Object {
            "x": "2015-01-02",
            "y": 2,
          },
          Object {
            "x": "2015-01-03",
            "y": 3,
          },
          Object {
            "x": "2015-01-04",
            "y": 5,
          },
        ]
      }
      hasLegend={false}
      hasToolTips={true}
      height={400}
      margin={
        Object {
          "bottom": 50,
          "left": 80,
          "right": 20,
          "top": 10,
        }
      }
      palette={
        Array [
          "color-chart-0",
          "color-chart-1",
          "color-chart-2",
          "color-chart-3",
          "color-chart-4",
          "color-chart-5",
        ]
      }
      width={1000}
      xAxisField="x"
      xAxisFormatter={[Function]}
      xAxisTickCount={null}
      xAxisTitle={null}
      xAxisTitleColor="#000"
      yAxisFields={
        Array [
          "y",
        ]
      }
      yAxisIsStacked={false}
      yAxisMin={0}
      yAxisTickCount={null}
      yAxisTitle={null}
      yAxisTitleColor="#000"
      yAxisTooltipFormatter={[Function]} />
    <LoadingMessage>
      <LoadingMessage.Icon>
        <LoadingIcon
          aspectRatio="xMidYMid meet"
          isBadge={false}
          size={16}
          speed="slow"
          viewBox="0 0 16 16" />
      </LoadingMessage.Icon>
      <LoadingMessage.Title>
        Custom Title
      </LoadingMessage.Title>
      <LoadingMessage.Body>
        Custom Body
      </LoadingMessage.Body>
    </LoadingMessage>
  </LoadingIndicator>
  <LoadingIndicator
    hasOverlay={true}
    isLoading={true}
    overlayKind="light">
    <BarChart
      data={
        Array [
          Object {
            "x": "2015-01-01",
            "y": 1,
          },
          Object {
            "x": "2015-01-02",
            "y": 2,
          },
          Object {
            "x": "2015-01-03",
            "y": 3,
          },
          Object {
            "x": "2015-01-04",
            "y": 5,
          },
        ]
      }
      hasLegend={false}
      hasToolTips={true}
      height={400}
      margin={
        Object {
          "bottom": 50,
          "left": 80,
          "right": 20,
          "top": 10,
        }
      }
      palette={
        Array [
          "color-chart-0",
          "color-chart-1",
          "color-chart-2",
          "color-chart-3",
          "color-chart-4",
          "color-chart-5",
        ]
      }
      width={1000}
      xAxisField="x"
      xAxisFormatter={[Function]}
      xAxisTickCount={null}
      xAxisTitle={null}
      xAxisTitleColor="#000"
      yAxisFields={
        Array [
          "y",
        ]
      }
      yAxisIsStacked={false}
      yAxisMin={0}
      yAxisTickCount={null}
      yAxisTitle={null}
      yAxisTitleColor="#000"
      yAxisTooltipFormatter={[Function]} />
    <LoadingMessage
      Body="Please wait"
      Icon={
        <LoadingIcon
          aspectRatio="xMidYMid meet"
          isBadge={false}
          size={16}
          speed="fast"
          viewBox="0 0 16 16" />
      }
      Title="Enhancing..." />
  </LoadingIndicator>
</div>
`;

exports[`LoadingIndicator [common] example testing should match snapshot for 4.no-title 1`] = `
<LoadingIndicator
  hasOverlay={true}
  isLoading={true}
  overlayKind="light">
  <BarChart
    data={
      Array [
        Object {
          "x": "2015-01-01",
          "y": 1,
        },
        Object {
          "x": "2015-01-02",
          "y": 2,
        },
        Object {
          "x": "2015-01-03",
          "y": 3,
        },
        Object {
          "x": "2015-01-04",
          "y": 5,
        },
      ]
    }
    hasLegend={false}
    hasToolTips={true}
    height={400}
    margin={
      Object {
        "bottom": 50,
        "left": 80,
        "right": 20,
        "top": 10,
      }
    }
    palette={
      Array [
        "color-chart-0",
        "color-chart-1",
        "color-chart-2",
        "color-chart-3",
        "color-chart-4",
        "color-chart-5",
      ]
    }
    width={1000}
    xAxisField="x"
    xAxisFormatter={[Function]}
    xAxisTickCount={null}
    xAxisTitle={null}
    xAxisTitleColor="#000"
    yAxisFields={
      Array [
        "y",
      ]
    }
    yAxisIsStacked={false}
    yAxisMin={0}
    yAxisTickCount={null}
    yAxisTitle={null}
    yAxisTitleColor="#000"
    yAxisTooltipFormatter={[Function]} />
  <LoadingMessage
    Title={null} />
</LoadingIndicator>
`;

exports[`LoadingIndicator [common] example testing should match snapshot for 5.no-overlay 1`] = `
<LoadingIndicator
  hasOverlay={false}
  isLoading={true}
  overlayKind="light">
  <BarChart
    data={
      Array [
        Object {
          "x": "2015-01-01",
          "y": 1,
        },
        Object {
          "x": "2015-01-02",
          "y": 2,
        },
        Object {
          "x": "2015-01-03",
          "y": 3,
        },
        Object {
          "x": "2015-01-04",
          "y": 5,
        },
      ]
    }
    hasLegend={false}
    hasToolTips={true}
    height={400}
    margin={
      Object {
        "bottom": 50,
        "left": 80,
        "right": 20,
        "top": 10,
      }
    }
    palette={
      Array [
        "color-chart-0",
        "color-chart-1",
        "color-chart-2",
        "color-chart-3",
        "color-chart-4",
        "color-chart-5",
      ]
    }
    width={1000}
    xAxisField="x"
    xAxisFormatter={[Function]}
    xAxisTickCount={null}
    xAxisTitle={null}
    xAxisTitleColor="#000"
    yAxisFields={
      Array [
        "y",
      ]
    }
    yAxisIsStacked={false}
    yAxisMin={0}
    yAxisTickCount={null}
    yAxisTitle={null}
    yAxisTitleColor="#000"
    yAxisTooltipFormatter={[Function]} />
</LoadingIndicator>
`;

exports[`LoadingIndicator [common] example testing should match snapshot for 6.light-overlay 1`] = `
<LoadingIndicator
  hasOverlay={true}
  isLoading={true}
  overlayKind="light">
  <BarChart
    data={
      Array [
        Object {
          "x": "2015-01-01",
          "y": 1,
        },
        Object {
          "x": "2015-01-02",
          "y": 2,
        },
        Object {
          "x": "2015-01-03",
          "y": 3,
        },
        Object {
          "x": "2015-01-04",
          "y": 5,
        },
      ]
    }
    hasLegend={false}
    hasToolTips={true}
    height={400}
    margin={
      Object {
        "bottom": 50,
        "left": 80,
        "right": 20,
        "top": 10,
      }
    }
    palette={
      Array [
        "color-chart-0",
        "color-chart-1",
        "color-chart-2",
        "color-chart-3",
        "color-chart-4",
        "color-chart-5",
      ]
    }
    width={1000}
    xAxisField="x"
    xAxisFormatter={[Function]}
    xAxisTickCount={null}
    xAxisTitle={null}
    xAxisTitleColor="#000"
    yAxisFields={
      Array [
        "y",
      ]
    }
    yAxisIsStacked={false}
    yAxisMin={0}
    yAxisTickCount={null}
    yAxisTitle={null}
    yAxisTitleColor="#000"
    yAxisTooltipFormatter={[Function]} />
</LoadingIndicator>
`;
